File: sample/04_cgi/www/cgi/hello.py
#!/usr/bin/python3
import os
import sys


def main():
    body = sys.stdin.read()
    method = os.environ.get('REQUEST_METHOD', '')
    qs = os.environ.get('QUERY_STRING', '')

    sys.stdout.write('Content-Type: text/plain\r\n')
    sys.stdout.write('\r\n')
    sys.stdout.write('python ok\n')
    sys.stdout.write('method=' + method + '\n')
    sys.stdout.write('query=' + qs + '\n')
    sys.stdout.write('body=' + body)


if __name__ == '__main__':
    main()

File: sample/04_cgi/www/cgi/hello.js
const fs = require('fs');

function readStdin() {
  try {
    return fs.readFileSync(0, 'utf8');
  } catch (e) {
    return '';
  }
}

const body = readStdin();
const method = process.env.REQUEST_METHOD || '';
const qs = process.env.QUERY_STRING || '';

process.stdout.write('Content-Type: text/plain\r\n');
process.stdout.write('\r\n');
process.stdout.write('node ok\n');
process.stdout.write('method=' + method + '\n');
process.stdout.write('query=' + qs + '\n');
process.stdout.write('body=' + body);
